File: app/src/main/java/com/kylemadsen/testandroid/animation/AnimationController.java
package com.kylemadsen.testandroid.animation;

import android.graphics.drawable.Animatable;
import android.view.View;
import android.widget.ImageView;

import com.kylemadsen.testandroid.R;
import com.kylemadsen.core.view.ViewController;
import com.kylemadsen.testandroid.utils.ViewUtensil;

public class AnimationController implements ViewController {

    @Override
    public int getLayoutId() {
        return R.layout.animation_view;
    }

    @Override
    public void attach(final View view) {
        final ImageView imageView = ViewUtensil.findById(view, R.id.image_view);
        ((Animatable) imageView.getDrawable()).start();
    }

    @Override
    public void detach() {

    }
}

File: app/src/main/java/com/kylemadsen/testandroid/animation/AnimationController.kt
package com.kylemadsen.testandroid.animation

import android.graphics.drawable.Animatable
import android.view.View
import android.widget.ImageView

import com.kylemadsen.testandroid.R
import com.kylemadsen.core.view.ViewController

class AnimationController : ViewController {

    override val layoutId: Int
        get() = R.layout.animation_view

    override fun attach(view: View) {
        val imageView: ImageView = view.find(R.id.image_view)
        (imageView.drawable as Animatable).start()
    }

    override fun detach() {

    }
}
